frameworklauncher/src/main/java/com/microsoft/frameworklauncher/applicationmaster/SelectionManager.java

```diff
@@ -181,7 +190,7 @@ public synchronized SelectionResult select(
 
       // Test Optimized Resource
       ResourceDescriptor optimizedRequestResource = YamlUtils.deepCopy(requestResource, ResourceDescriptor.class);
-      if (requestResource.getGpuAttribute() == 0) {
+      if (requestResource.getGpuAttribute() == 0 && requestResource.getGpuNumber() > 0) {
```

Comment:
>requestResource.getGpuAttribute() == 0 && requestResource.getGpuNumber() > 0 [](start = 10, length = 76)

what if "(requestResource.getGpuAttribute() == 0 && requestResource.getGpuNumber() == 0)"? ignore?

Reply:
If(getGpuNumber() == 0), we won't do local GPU selection.  If(getGpuAttribute() > 0), means use specified GPU, we also ignore local GPU selection. 

---
In reply to: [166908987](https://github.com/Microsoft/pai/pull/318#discussion_r166908987) [](ancestors = 166908987)